x/auth/keeper/deterministic_test.go

```diff
@@ -0,0 +1,112 @@
+package keeper_test
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/testutil"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"pgregory.net/rapid"
+)
+
+type DeterministicTestSuite struct {
+	suite.Suite
+
+	ctx sdk.Context
+
+	queryClient   types.QueryClient
+	accountKeeper keeper.AccountKeeper
+	encCfg        moduletestutil.TestEncodingConfig
+}
+
+func TestDeterministicTestSuite(t *testing.T) {
+	suite.Run(t, new(DeterministicTestSuite))
+}
+
+func (suite *DeterministicTestSuite) SetupTest() {
+	suite.encCfg = moduletestutil.MakeTestEncodingConfig(auth.AppModuleBasic{})
+
+	key := sdk.NewKVStoreKey(types.StoreKey)
+	testCtx := testutil.DefaultContextWithDB(suite.T(), key, sdk.NewTransientStoreKey("transient_test"))
+	suite.ctx = testCtx.Ctx.WithBlockHeader(tmproto.Header{})
+
+	maccPerms := map[string][]string{
+		"fee_collector":          nil,
+		"mint":                   {"minter"},
+		"bonded_tokens_pool":     {"burner", "staking"},
+		"not_bonded_tokens_pool": {"burner", "staking"},
+		multiPerm:                {"burner", "minter", "staking"},
+		randomPerm:               {"random"},
+	}
+
+	suite.accountKeeper = keeper.NewAccountKeeper(
+		suite.encCfg.Codec,
+		key,
+		types.ProtoBaseAccount,
+		maccPerms,
+		"cosmos",
+		types.NewModuleAddress("gov").String(),
+	)
+
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.encCfg.InterfaceRegistry)
+	types.RegisterQueryServer(queryHelper, suite.accountKeeper)
+	suite.queryClient = types.NewQueryClient(queryHelper)
+}
+
+func (suite *DeterministicTestSuite) runIterations(addr sdk.AccAddress, prevRes types.AccountI, checkPrevRes bool) {
```

Comment:
We don't need this I think:

```suggestion
func (suite *DeterministicTestSuite) runIterations(addr sdk.AccAddress, prevRes types.AccountI) {
```